vm/src/pyobject.rs

```diff
@@ -1562,16 +1558,31 @@ impl PyObject {
     }
 
     pub fn payload<T: PyValue>(&self) -> Option<&T> {
-        self.payload.downcast_ref()
+        let payload: &dyn Any = &self.payload;
+        payload.downcast_ref()
     }
 }
 
-// The intention is for this to replace `PyObjectPayload` once everything is
-// converted to use `PyObjectPayload::AnyRustvalue`.
-pub trait PyValue: Any + fmt::Debug {
+pub trait PyValue: fmt::Debug + 'static {
     fn required_type(ctx: &PyContext) -> PyObjectRef;
 }
 
+pub trait PyValuePayload: Any + fmt::Debug + 'static {
```

Comment:
For the name, how about `PyObjectPayload` - @cthulahoops and @windelbouwman both preferred that, and since this separates it from `PyValue`, it resolves my concerns about `PyObjectPayload` being an unintuitive name from an API perspective.

Reply:
Sounds good!